app/src/main/java/com/pvp/app/ui/screen/goals/GoalCreation.kt

```diff
@@ -328,56 +337,41 @@ fun StepPicker(
 }
 
 @Composable
-private fun StepSelector(
-    isSelected: Boolean,
-    onClick: () -> Unit
+fun StepSelector(
+    selectedStepType: StepType,
+    onStepTypeChange: (StepType) -> Unit
 ) {
-    Row(
-        horizontalArrangement = Arrangement.SpaceBetween,
+    val selectedTabIndex = when (selectedStepType) {
+        StepType.Steps -> 0
+        StepType.Distance -> 1
+    }
+
+    PrimaryTabRow(
+        containerColor = MaterialTheme.colorScheme.secondaryContainer,
+        divider = {},
         modifier = Modifier
-            .background(
-                MaterialTheme.colorScheme.surfaceContainer,
-                MaterialTheme.shapes.medium
-            )
             .fillMaxWidth()
-            .padding(vertical = 4.dp)
+            .clip(MaterialTheme.shapes.medium),
+        selectedTabIndex = selectedTabIndex
     ) {
-        Box(
-            contentAlignment = Alignment.Center,
-            modifier = Modifier
-                .weight(1f)
-                .fillMaxWidth()
-                .height(40.dp)
-                .clickable { onClick() }
-                .background(
-                    if (isSelected) {
-                        MaterialTheme.colorScheme.secondaryContainer
-                    } else {
-                        Color.Transparent
-                    },
-                    MaterialTheme.shapes.medium
-                )
-        ) {
-            Text(text = "Steps")
-        }
-
-        Box(
-            contentAlignment = Alignment.Center,
-            modifier = Modifier
-                .weight(1f)
-                .fillMaxWidth()
-                .height(40.dp)
-                .clickable { onClick() }
-                .background(
-                    if (!isSelected) {
-                        MaterialTheme.colorScheme.secondaryContainer
-                    } else {
-                        Color.Transparent
-                    },
-                    MaterialTheme.shapes.medium
+        StepType.values().forEach { stepType ->
+            Tab(
+                modifier = Modifier.height(32.dp),
+                selected = selectedStepType == stepType,
+                onClick = { onStepTypeChange(stepType) }
+            ) {
+                Text(
+                    text = stepType.displayName,
+                    color = MaterialTheme.colorScheme.inverseSurface,
+                    style = MaterialTheme.typography.titleMedium,
+                    fontWeight = if (stepType == selectedStepType) FontWeight.Bold else FontWeight.Normal
                 )
-        ) {
-            Text(text = "Distance")
+            }
         }
     }
+}
+
+enum class StepType(val displayName: String) {
```

Comment:
`DistanceType` probably would be the better name choice. Should rename associated parameters/variables above too.